src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -17,18 +34,35 @@ public string Serialize(object sourceObject)
         {
             if (sourceObject == null)
             {
-                return String.Empty;
+                return string.Empty;
             }
 
-            var formatter = new BinaryFormatter();
+            dynamic serializedObject = (sourceObject is string)
+                ? sourceObject
+                : this.AddTypeInformation(sourceObject);
+
+            var json = SimpleJson.SerializeObject(serializedObject);
+            return Convert.ToBase64String(Encoding.UTF8.GetBytes(json));
+        }
 
-            using (var outputStream = new MemoryStream())
+        private dynamic AddTypeInformation(object sourceObject)
+        {
+            var sourceType = this.assemblyCatalog.GetAssemblies().Select(assembly => assembly.GetType(sourceObject.GetType().FullName)).FirstOrDefault(type => type != null);

```

Comment:
Well if you hadn't insisted on one commit message per PR then I might have
something that jogs my memory why I added it.

Remove it, run the tests and if all is ok then consider me told

On Tuesday, 23 February 2016, Kristian Hellang <notifications@github.com>
wrote:

> In src/Nancy/DefaultObjectSerializer.cs
> <https://github.com/NancyFx/Nancy/pull/2264#discussion_r53822433>:
>
> >
> > -            using (var outputStream = new MemoryStream())
> > +        private dynamic AddTypeInformation(object sourceObject)
> > +        {
> > +            var sourceType = this.assemblyCatalog.GetAssemblies().Select(assembly => assembly.GetType(sourceObject.GetType().FullName)).FirstOrDefault(type => type != null);
>
> You can't *think*, you need to *know*.
>
> The only way AssemblyQualifiedName returns null is if it's a generic type
> parameter. From MSDN:
>
> The assembly-qualified name of the Type, which includes the name of the
> assembly from which the Type was loaded, or null if the current instance
> represents a generic type parameter.
>
> [image: image]
> <https://cloud.githubusercontent.com/assets/582487/13261618/49c94f20-da61-11e5-8eb1-937c659111c6.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NancyFx/Nancy/pull/2264/files#r53822433>.
>